Implement combat

Aaahtemis opened the pull request:
Modified Program.cs and MainMenu.cs to integrate simplified combat mechanics.
Adjusted globals methods to allow padding and reset player input.
Combat is now playable with easily modifiable variables.

BadCoderName1 commented:
Code is easy to read
Contains comments
Variables naming is good
No repeated code
Method names good
Code fulfills what is described

- One thing I noticed is that certain rooms (toys and camping are the ones I tried) threw me an error before getting into combat. Clothing let me get into the combat but when the enemy reaches 0hp - the combat doesn't exit and no way to return.
(Just double check this and then happy to push it through)

Aaahtemis commented:
Reviewed and adjusted code to work in the scenarios (*of those that I'm aware of)
Fixed looping enemy turns.
Simplified inventory a little more.
Concerned about the fact that regardless of whether you win or lose, that data isn't stored and so you get the item anyway.

Wheerh1 commented:
Works as far as I quickly tested
code is well commented and easy enough to read
variables still well named and following conventions
no noted repeating code
methods are generally unbloated, the perform action method might be able to have hail mary and the 
"if (damageDealt > 0) // possibly add contingency for multiple attacked enemies and simplify code further." 
section moved to their own methods if we want even speedier readability.
does what is described in the task.

also like the colour that has been added to the different attacks, nice touch!

kavarnfrancis reviewed (approved):
Completely happy to approve this code per my comment in pull request. :)
